app/Domain/User/Service/UserCreator.php

```diff
@@ -22,7 +23,8 @@ public function __construct(private UserCreatorRepository $repository)
     public function createUser(CreateUserDTO $userDTO): int
     {
         $this->validateUser($userDTO);
-        return $this->repository->insert($userDTO);
+        $user = new User($userDTO->name, $userDTO->email, $userDTO->password);
+        return $this->repository->insert($user);
     }
 
     /**
```

Comment:
sei que não é desta sessão mas         if (is_null($userDTO->email)) {
            $errors['email'] = 'Input required';
        } elseif (filter_var($userDTO->email, FILTER_VALIDATE_EMAIL) === false) {
            $errors['email'] = 'Invalid email address';
        }

não poderia ser uma validação somente, ou retirar esse elseif, pois ele não me soa bem 😄 
sugestão apenas.

Reply:
Sim sim, to pensando numa estrturura um pouco melhor pra esa validação